service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/AllocationDecisionController.java

```diff
@@ -59,4 +59,31 @@ public AboutToStartOrSubmitCallbackResponse checkIfAllocationProposalIsMissing(
     private String checkIfAllocationProposalIsPresent(AllocationProposal data) {
         return data != null && StringUtils.isNotEmpty(data.getProposal()) ? "Yes" : "No";
     }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        AllocationDecision allocationDecision = caseData.getAllocationDecision();
+
+        AllocationDecision.AllocationDecisionBuilder decisionBuilder = ofNullable(allocationDecision)
+            .map(AllocationDecision::toBuilder)
+            .orElse(AllocationDecision.builder());
+
+        if (caseData.getAllocationDecision().getProposal() == null) {
+            decisionBuilder.proposal(caseData.getAllocationProposal().getProposal());
+            Map<String, Object> data = caseDetails.getData();
+            data.put("allocationDecision", decisionBuilder.build());
+        } else {
```

Comment:
Hmm, do we need this else statement at all?

Reply:
I removed it and the tests still pass :). So either issue with the test, or the else is not required, please check. Doesn't look like it is required.

Reply:
If it is necessary it could be simplified by inverting the if to check for not null and then just performing the `decisionBuilder.proposalReason(...)` inside the if. The other 3 lines appear in both  parts of the control statement and can therefore just occur after.

Reply:
discussed with both people and resolved

Reply:
discussed with guys and changed